### PR TITLE
refactor(kit): drop `async` pipe and `let` directive

### PR DESCRIPTION
**Pull request by splincode**
Fix for flaky
image

### CODE REVIEW

File: projects/kit/components/tree/components/tree/tree.component.ts [hunk 1]
```diff
@@ -48,10 +40,12 @@ export class TuiTreeComponent<T> implements DoCheck {
     @ViewChild(forwardRef(() => TuiTreeComponent))
     protected readonly child?: TuiTreeComponent<T>;
 
-    protected readonly children$ = this.check$.pipe(
-        startWith(null),
-        map(() => this.handler(this.value)),
-        distinctUntilChanged(),
+    protected readonly children = toSignal(
```
Comment (waterplea) [inline projects/kit/components/tree/components/tree/tree.component.ts:43]: `toSignal` immediately subscribes and at that time there's no `value`. I have a feeling this changes probably breaks something.

Comment (splincode) [reply projects/kit/components/tree/components/tree/tree.component.ts:43]: Don't worry, be happy :)

**Comment by mdlufy** [comment]
> Fix for image

@splincode why tests not failed, if it fixes anything case?

**Comment by splincode** [comment]
@mdlufy flaky

File: projects/kit/components/tree/components/tree/tree.component.ts (hunk 1 above)
Comment (waterplea) [reply projects/kit/components/tree/components/tree/tree.component.ts:43]: Unless we can be sure, I don't think we should merge that.

**Pull request closed without merging**